crates/ty_server/src/server/api/diagnostics.rs

```diff
@@ -113,6 +114,69 @@ pub(super) fn publish_diagnostics(
     Ok(())
 }
 
+/// Publishes settings diagnostics for all the projects using the [publish diagnostics
+/// notification].
+///
+/// [publish diagnostics notification]: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_publishDiagnostics
+pub(super) fn publish_settings_diagnostics(session: &mut Session, client: &Client) {
```

Comment:
I wonder if we should change this method to take a `&mut Project` instead. It should be unlikely that a user changes all projects at once. It's more likely that they changed a setting for a specific project. 

This would also ensure that we don't publish diagnostics for the default database (it should never have setting diagnostics but initializing it isn't entirely free)

Reply:
`did_change_watched_files` does group all files by database and then calls `apply_changes` exacltly once per database. See this loop where `root `maps to the root of a database:


https://github.com/astral-sh/ruff/blob/fe8f0cfcf9760d39aa716deb180678bd72cd3aba/crates/ty_server/src/server/api/notifications/did_change_watched_files.rs#L88-L94

This should allow you to make `publish_settings_diagnostics` per database.